agent/Cargo.toml

```diff
@@ -25,9 +25,9 @@ futures-core = "0.3"
 futures-util = "0.3"
 hyper = "0.14.2"
 h2 = "=0.3.3"
-kube = { version = "0.59.0",  features = ["derive"] }
-kube-runtime = "0.59.0"
-k8s-openapi = { version = "0.13.0", default-features = false, features = ["schemars", "v1_16"] }
+kube = { version = "0.69.1",  features = ["derive"] }
+kube-runtime = "0.69.1"
+k8s-openapi = { version = "0.14.0", default-features = false, features = ["schemars", "v1_22"] }
```

Comment:
Looks like each [feature corresponds to a supported K8s version](https://github.com/Arnavion/k8s-openapi/blob/master/Cargo.toml#L54). I wonder if we want to list all from 1_16 to 1_22? curious how that changes the size of the crate. The e2e tests did pass on all the versions curiously.

Reply:
Should we do `1_23` instead of `1_22`?

Reply:
It looks like based on how they are pulling K8s [swaggers here](https://github.com/Arnavion/k8s-openapi/blob/563883804ca06448f24890ebf47b4f7efa491caf/k8s-openapi-codegen/src/supported_version.rs#L60) that whatever version feature is chosen is the K8s client specification that is used (kubernetes source APIs can be found [here](https://github.com/kubernetes/kubernetes/tree/v1.23.4/api/openapi-spec)). I think it is safe to go with `1_22` or better yet `1_23` but i would be curious how to determine when the API we are using is not backwards compatible.

Reply:
If I follow that kubernetes api doc you linked through a few more hops I eventually end up at the Open API Spec. The [versioning clause states](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#versions):

> The OpenAPI Specification is versioned using a major.minor.patch versioning scheme.

which would make me think that so long as it doesnt jump a major version we would be fine. However the spec goes on to say:

> Occasionally, non-backwards compatible changes may be made in minor versions of the OAS where impact is believed to be low relative to the benefit provided.

which is a little spooky. I'm asking the maintainer of the crate for their thoughts [here](https://github.com/Arnavion/k8s-openapi/issues/113) . However I'd imagine the responsibility eventually falls to the end client code (so Akri) to handle the regression tests from changes in the OAPI. In that regard, if Akri's regression tests succeed then I imagine we would be fine.

Reply:
in any case though i've bumped the version to to v1_23. The local build stuff passed locally, so we'll see what the CI checks say.

Reply:
Thanks for inquiring about this! I think we are all set to merge.

Reply:
Cool, I agree. And to repost what is probably the relevant situation in that conversation here:

> "Situation 3: Kubernetes releases v1.61.0. There will be a new major release of k8s-openapi, ie v0.101.0, with support for this new v1.61 line. This is because adding support for new Kubernetes version lines is semver-incompatible"...

So unless we have a major version change for k8s-openapi its fine. 

Sorry for the delay on this response. It's been an interesting week for me.